lib/web_ui/lib/src/engine/platform_dispatcher/view_focus_binding.dart

```diff
@@ -67,36 +62,58 @@ final class ViewFocusBinding {
   });
 
   void _handleFocusChange(DomElement? focusedElement) {
+    final int? lastViewId = _lastViewId;
     final int? viewId = _viewId(focusedElement);
-    if (viewId == _lastViewId) {
+    if (viewId == lastViewId) {
       return;
     }
+
     final ui.ViewFocusEvent event;
-    if (viewId == null) {
-      event = ui.ViewFocusEvent(
-        viewId: _lastViewId!,
-        state: ui.ViewFocusState.unfocused,
-        direction: ui.ViewFocusDirection.undefined,
-      );
-    } else {
+    if (viewId != null) {
       event = ui.ViewFocusEvent(
         viewId: viewId,
         state: ui.ViewFocusState.focused,
         direction: _viewFocusDirection,
       );
+    } else {
+      event = ui.ViewFocusEvent(
+        viewId: lastViewId!,
+        state: ui.ViewFocusState.unfocused,
+        direction: ui.ViewFocusDirection.undefined,
+      );
     }
     _lastViewId = viewId;
-    _notify(event);
+    _markViewAsFocusable(lastViewId);
+    _markViewAsFocusable(viewId, reachableByKeyboard: false);
+    _onViewFocusChange(event);
```

Comment:
WHAT IF...

The `ViewFocusBinding` class exposed a:

```
Stream<ViewFocusEvent> onFocusChanged;
```

And users from outside (say, the PlatformDispatcher) could subscribe to that stream, as needed?

Internally, you could have a StreamController that lets you check when the stream is listened/unlistened to, and then attach/detach the DOM events as needed (similar to what you had before with the array of listeners!)

That way, this class doesn't need to keep a copy of the `_onViewFocusChange` handler, AND potentially multiple classes could listen to the ViewFocusChange events for their own purposes (say one for the framework, and another for debugging)

((PS: This also solves the problem that we have with the `KeyboardBinding` not exposing the events that we might need elsewhere in the engine, and instead being very rigid in what their callback/handling of events is))

Reply:
Great advice @ditman. What a true pro! 

Done.